Clarification of the behavior of calibraty_linear filter (#1701)
OttoWinter/esphomedocs: components/sensor/index.rst

The arguments are a list of data points, each in the form ``MEASURED -> TRUTH``. ESPHome will
then fit a linear equation to the values (using least squares). So you need to supply at least

> two values. If more than two points are specified, they may not lie on a straight line. In this 
> case, the calculated results often do not correspond to the specified values. Especially for 
> non-linear relationships, the result can deviate strongly from the expected value. It is 
> recommended to use the filter calibrate linear only if two given data points are sufficient.

Reviewer: ```suggestion
two values. If more than two points are specified, the line of best fit may not pass through any of
the specified values, and the result may be very different from the expected value. Therefore, it is
recommended to use the filter calibrate linear only if two given data points are sufficient.
```

What do you think about this phrasing?

Reviewer: Oh! I like the "if more than two values are given a linear solution will be calculated and will not represent each value exactly", I didn't re-read it when I first came back to this.

How about

```suggestion
two values. If more than two values are given a linear solution will be calculated and may not
represent each value exactly. It is recommended to use the filter calibrate linear only if two data
points are sufficient.
```

Reviewer 2: > It is recommended to use the filter calibrate linear only if two data points are sufficient.

Maybe it's my physics background, but I don't fully agree with this. You should only use `calibrate_linear` if the data you're trying to calibrate (and thus the data points you've sampled) is linear, but it's absolutely fine to use more than two data points.

Reviewer: Removed that sentence here: https://github.com/esphome/esphome-docs/pull/1830

The sentence explaining that it won't always pass through both points has been left in.

Reviewer 2: Excellent.
